pkg/config/node.go

```diff
@@ -306,23 +310,105 @@ func GetIngressConfig(kubeconfigPath string, filterIpType string) (ingressConfig
 		return ingressConfig, err
 	}
 
+	if len(vips) == 0 {
+		// This is not necessarily an error path because in handleBootstrapStopKeepalived we do
+		// call this function without providing any VIPs. Because of this, we only want to mark
+		// this scenario and avoid trying to calculate the machine networks.
+		log.Infof("Requested GetIngressConfig for empty VIP list.")
+	} else {
+		// As it is not possible to get cluster's Machine Network directly, we are using a workaround
+		// by detecting which of the local interfaces belongs to the same subnet as requested VIP.
+		// This interface can be used to detect what was the original machine network as it contains
+		// the subnet mask that we need.
+		machineNetwork, err = utils.GetLocalCIDRByIP(vips[0])
+		if err != nil {
+			return ingressConfig, err
+		}
+	}
+
 	for _, node := range nodes.Items {
-		for _, address := range node.Status.Addresses {
-			if address.Type == v1.NodeInternalIP {
-				if filterIpType != "" {
-					if (net.ParseIP(filterIpType).To4() != nil && net.ParseIP(address.Address).To4() == nil) ||
-						(net.ParseIP(filterIpType).To4() == nil && net.ParseIP(address.Address).To4() != nil) {
-						continue
-					}
-				}
-				ingressConfig.Peers = append(ingressConfig.Peers, address.Address)
-			}
+		addr, err := getNodeIpForRequestedIpStack(node, vips, machineNetwork)
+		if err != nil {
+			log.Warnf("For node %s could not retrieve node's IP. Ignoring", node.ObjectMeta.Name)
+		} else {
+			ingressConfig.Peers = append(ingressConfig.Peers, addr)
 		}
 	}
 
 	return ingressConfig, nil
 }
 
+func getNodeIpForRequestedIpStack(node v1.Node, filterIps []string, machineNetwork string) (string, error) {
+	log.SetLevel(logrus.DebugLevel)
```

Comment:
If we're forcing this on, should these log messages just be Info level?

Reply:
Sure, just changed. I kind of thought about a future when we want to have an env variable to control the log level to decrease verbosity, but it will be simple to then lower level from info to debug